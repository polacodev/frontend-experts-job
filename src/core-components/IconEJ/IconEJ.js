import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Icon } from 'native-base';

import styles from './IconEJ.style';

const IconEJ = ({
  iconName = '',
  color = '',
  size = 20,
  onPressIcon = () => ({}),
}) => {
  return onPressIcon.name ? (
    <TouchableOpacity testID="button-icon" onPress={onPressIcon}>
      <Icon
        type="FontAwesome"
        name={iconName}
        style={styles(color, size).icon}
      />
    </TouchableOpacity>
  ) : (
    <Icon type="FontAwesome" name={iconName} style={styles(color, size).icon} />
  );
};

export default IconEJ;
