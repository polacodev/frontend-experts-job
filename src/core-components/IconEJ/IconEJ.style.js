import { StyleSheet } from 'react-native';

export const styles = (color, size) =>
  StyleSheet.create({
    icon: {
      fontSize: size,
      color: color,
    },
  });

export default styles;
