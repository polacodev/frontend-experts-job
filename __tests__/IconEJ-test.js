import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

import IconEJ from '../src/core-components/IconEJ/IconEJ';
jest.useFakeTimers();
describe('<IconEJ />', () => {
  const user = {
    name: 'Brian',
    lastName: 'Stobbe',
    workarea: 'Developer',
    cellphone: '67415847',
  };
  test('renders <IconEJ /> correctly', () => {
    const tree = renderer
      .create(
        <IconEJ
          iconName="times-circle-o"
          color="#ffffff"
          size={20}
          onPressIcon={() => ({})}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call onEventMock when the icon is pressed', async () => {
    const onEventMock = jest.fn();

    const { findByTestId } = render(
      <IconEJ
        iconName="times-circle-o"
        color="#ffffff"
        size={20}
        onPressIcon={() => onEventMock(user)}
      />,
    );
    fireEvent.press(await findByTestId('button-icon'), onEventMock);
    expect(onEventMock).toHaveBeenCalled();
  });

  test('should return undefined due the empty props', () => {
    const onEventMock = jest.fn();
    const { container } = render(
      <IconEJ
        iconName="times-circle-o"
        color="#ffffff"
        size={20}
        onPressIcon={() => onEventMock(user)}
      />,
    );
    expect(container.firstChild).toBeUndefined();
  });
});
