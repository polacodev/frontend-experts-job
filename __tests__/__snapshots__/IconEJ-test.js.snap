// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconEJ /> renders <IconEJ /> correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="button-icon"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#ffffff",
            "fontSize": 20,
          },
        ],
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    type="FontAwesome"
  >
    
  </Text>
</View>
`;
